tensorboard/webapp/metrics/store/metrics_store_internal_utils.ts

```diff
@@ -428,4 +451,50 @@ function getSelectedSteps(selectedTime: LinkedTime | null, steps: number[]) {
   return selectedStepsInRange;
 }
 
-export const TEST_ONLY = {getImageCardStepValues, getSelectedSteps, util};
+/**
+ * Gets cardStepIndex updated based on single selection.
+ */
+function getNextCardStepIndexOnSingleSelection(
+  cardId: string,
+  startStep: number,
+  steps: number[],
+  previousCardStepIndex: CardStepIndexMap
+) {
+  const nextCardStepIndex = {...previousCardStepIndex};
```

Comment:
I find it odd that you're regenerating nextCardStepIndex over and over - it doesn't seem like it should be the responsibility of this function - is it necessary?  I think the only thing that should be generating it is generateNextCardStepIndexFromSelectedTime. Consider having this function return the index value and have it updated by generateNextCardStepIndexFromSelectedTime.

In that case you would also have two choices: When you can't calculate a new card index, do you return the old one or do you return null? If null, the call to this function could look something like:

```
    if (selectedTime.end === null) {
      const cardStep = getImageCardStepFromSingleSelection(
        selectedTime.start.step,
        steps
      );
      if (nextCardStep !== null) {
          nextCardStepIndex[cardId] = nextCardStep;
      }
    }
```

If this all makes sense then I'd also recommend changing the name. Add 'Image', remove 'Index', use same verb 'From' as the parent function. 

Something like: getImageCardStepFromSingleSelection.

Reply:
Thanks for the comment.  Updated to `getImageCardStepIndexFromSingleSelection`
Still prefer reserving the 'index' part. Without index it sound like stepValues (which all renamed to steps).

What do you think?

Reply:
Thanks for the comment. Updated the method to return step index.
Still prefer keep 'index' otherwise it's confusing with stepValues, which all renames to steps.
steps could be [193,000, 283,000, 989,000] while the step index is [0, 1, 2]